### PR TITLE
Update onboarding task documentation & add back `example` command

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Related https://github.com/woocommerce/woocommerce/issues/33293

Update onboarding task documentation 

1. Add back `pnpm --filter=@woocommerce/admin-library example` command so that we can build the examples.
2. Update the example's webpack config to ignore the output directory to avoid infinite rebuilds
3. Improve onboarding task documentation
     - Add readme to add task example
     - Update `docs/extension-development/handling-merchant-onboarding.md` (It's outdated.)

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Review the documentation changes.
2. Run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example`
3. Confirm `plugins/add-task` is created.
4. Activate the plugin and confirm the task is added to the list.
5. Run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example --watch`
6.  Confirm it watches for changes and only rebuilds when the file changes.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Update onboarding task documentation and fix example command

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/docs/examples/extensions/examples.config.js [hunk 1]
```diff
@@ -103,6 +103,12 @@ const webpackConfig = {
 		} ),
 		new WooCommerceDependencyExtractionWebpackPlugin(),
 	],
+	watchOptions: {
+		ignored: [
+			path.resolve( __dirname, `../../../../${ extension }/` ),
```
Comment (psealock) [inline plugins/woocommerce-admin/docs/examples/extensions/examples.config.js]: When I change any file, say `plugins/add-task/js/index.js` it doesn't rebuild. Could this `ignore` be causing that?

**Review by psealock — COMMENTED** [review]
Hi @chihsuan, this is working well. I wasn't able to get the example plugin to rebuild though.

File: plugins/woocommerce-admin/docs/examples/extensions/examples.config.js (hunk 1 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/docs/examples/extensions/examples.config.js]: Nope, this is unrelated but I've just changed it to ignore `dist` folder only.

**Comment by chihsuan** [comment]
> Hi @chihsuan, this is working well. I wasn't able to get the example plugin to rebuild though.

Hey @psealock Good catch! I found that it requires saving twice to refresh because [`copy-webpack-plugin`](https://webpack.js.org/plugins/copy-webpack-plugin/) is **not designed to copy files generated from the build process**.

I changed the output path to the path under the plugin folder in b3ac4253fd518899690f66f395a6945174e58f8e to fix it. Could you please try again? You might need to remove `plugins/woocommerce-admin/docs/examples/extensions/add-task/dist` folder first before testing. 🙏

File: plugins/woocommerce-admin/docs/examples/extensions/add-task/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+After running the command above and activating the plugin, you should see a new task in the onboarding wizard:
+
+<!-- markdownlint-disable-next-line no-inline-html -->
+<img src="./images/task-example.png" width="500px" alt="Screenshot of the onboarding wizard with the custom task" />
+
+To watch for changes in the example plugin and rebuild it automatically, run the following command:
+
+```bash
+WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example --watch
+```
```
Comment (ilyasfoo) [inline plugins/woocommerce-admin/docs/examples/extensions/add-task/README.md:24]: When I initially tried this, I modified the built file in `plugins/add-task/js/index.js`, which did nothing. Can we also add the detail on where to make the changes? I think it's an important detail to add. What do you think?

**Review by ilyasfoo — APPROVED** [review]
Thanks, @chihsuan! Tested well, but I have a minor comment. Pre-approving

File: plugins/woocommerce-admin/docs/examples/extensions/add-task/README.md (hunk 2 above)
Comment (chihsuan) [reply plugins/woocommerce-admin/docs/examples/extensions/add-task/README.md:24]: Sounds good! Added more details in ca5a1ba6d7b40cf371f0d8167ee9bcaf3cee0652

**Comment by psealock** [comment]
Thanks for the changes, when I run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example` I get the following error, look familiar?

```
ERROR in Conflict: Multiple assets emit different content to the same filename index.js.map
```

**Comment by chihsuan** [comment]
> Thanks for the changes, when I run `WC_EXT=add-task pnpm --filter=@woocommerce/admin-library example` I get the following error, look familiar?
> 
> ```
> ERROR in Conflict: Multiple assets emit different content to the same filename index.js.map
> ```

@psealock Yes, did you remove the `plugins/woocommerce-admin/docs/examples/extensions/add-task/dist` folder? 🤔

**Review by psealock — APPROVED** [review]
Thanks for the clarification @chihsuan, this is working well now 🚢

**Pull request merged**